files/nrpe/check_nova_neutron_order.sh

```diff
@@ -0,0 +1,47 @@
+#!/bin/bash
+#
+# Rudimentary check for monitoring that neutron rules are at the very top
+# of iptables ordering.
+
+set -e
+
+STATE_OK=0
+STATE_WARNING=1
+STATE_CRITICAL=2
+STATE_UNKNOWN=3
+
+usage ()
+{
+    echo "Usage: $0 [OPTIONS]"
+    echo " -h               Get help"
+}
+
+while getopts 'h:' OPTION
+do
+    case $OPTION in
+        h)
+            usage
+            exit ${STATE_WARNING}
+            ;;
+        \?)
+            usage
+            exit ${STATE_WARNING}
+            ;;
+        *)
+            usage
+            exit ${STATE_WARNING}
+            ;;
+    esac
+done
+shift "$((OPTIND-1))"
+
+IPTABLES_NEUTRON_STATUS="$(iptables -L|grep -E '^nova-filter-top|^neutron-filter-top'|head -1|grep neutron)"
```

Comment:
I think you must add `--numeric` to the `iptables` command to make the script run in a reasonable amount of time on production oversub nodes, otherwise iptables will look up all host names for all IPs.

While you are at it, I guess it is good practice to use the long flag names in shell scripts. `-L` --> `--list`

Reply:
Good point. I have ran this as is from my laptop as a one-off, and the runtime has been well below 1 minute even without numeric. But can see the numeric as a must have for something that is run periodically. Should be fixed now.